### DISCUSSION TITLE
It would be nice to have a 3.6.2 .deb package

### DISCUSSION THREAD

**Discussion by bspratt** [jamulussoftware/jamulus#1449] [Bug Investigation]
A lot of recent server updates were likely performed using Simon Tomlinson's script (it's what I recommend in my Vultr setup guide https://m.facebook.com/notes/jamulus-worldjam/howto-vultr-edition-of-idiots-guide-to-setting-up-a-jamulus-server/378554646728489/). Simon's script was changed for 3.7.0 to use packages rather than building from source, which makes sense, but that makes it tricky to roll back to an earlier version built from source.

Obviously we want people moving forward, not backward, but there's a suspicion out there that 3.7.0 might be not quite right yet (I'm seeing unexplained jamulus server restarts, for example, but can't swear it's a 3.7 thing). Providing a downgrade path will at least allow folks to convince themselves that a downgrade isn't actually helpful.

**Comment by gene96817** [comment]
The ability to rollback the release will also reduce the hesitancy of moving to the latest release. A lot of users will not upgrade for a good number of months as they wait to hear of problems. They don't want to be trapped with a new release with problems. Easy rollback will make it a low risk to try the new release.

**Reply by dcorson-ticino-com** [reply]
Does it no longer work using the original script for installing 3.6.2 ?
I don't see the problem.

**Reply by bspratt** [reply]
>> Does it no longer work using the original script for installing 3.6.2 ?

The 3.6.2 era version of Simon's script builds from source. The current version installs a package, because not all server machines have enough resources to build from source. So to reliably roll back we'd need a 3.6.2 package.

Also, the two script versions install to different locations, so just grabbing an old copy of Simon's script won't help because the current service file won't look for the 3.6.2 build.

**Comment by hoffie** [comment]
I have just tried blindly backporting the autobuild logic to 3.6.2 and pushing a tag in my fork (as I don't want to make this sound official in any way). It seems to have worked for the Debian build, but I haven't tried at all:

https://github.com/hoffie/jamulus/actions/runs/717580913

I agree that if there are any issues preventing upgrading to 3.7.0, they should be documented and fixed.

**Reply by bspratt** [reply]
Agreed, any actual problems need to be documented and fixed. I'll do what I can on that front by updating a different less critical server to 3.7.0.
